test/jdk/java/awt/font/FontScaling/StretchedFontTest.java

```diff
@@ -0,0 +1,196 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import javax.imageio.ImageIO;
+
+import static java.awt.GraphicsEnvironment.getLocalGraphicsEnvironment;
+import static java.awt.RenderingHints.KEY_TEXT_ANTIALIASING;
+import static java.awt.RenderingHints.VALUE_TEXT_ANTIALIAS_LCD_HRGB;
+import static java.awt.RenderingHints.VALUE_TEXT_ANTIALIAS_OFF;
+import static java.awt.RenderingHints.VALUE_TEXT_ANTIALIAS_ON;
+import static java.awt.image.BufferedImage.TYPE_3BYTE_BGR;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8312555
+ * @summary Verifies that hieroglyphs are stretched by AffineTransform.scale(2, 1)
+ * @run main StretchedFontTest
+ */
+public class StretchedFontTest {
+    private static final String TEXT = "\u6F22";
+    private static final int FONT_SIZE = 20;
+
+    private static final AffineTransform STRETCH_TRANSFORM =
+            AffineTransform.getScaleInstance(2.0, 1.0);
+
+    public static void main(String[] args) {
+        List<String> errors =
+                Arrays.stream(getLocalGraphicsEnvironment()
+                              .getAvailableFontFamilyNames(Locale.US))
+                      .map(family -> new Font(family, Font.PLAIN, FONT_SIZE))
+                      .filter(font -> font.canDisplay(TEXT.codePointAt(0)))
+                      .map(font -> font.deriveFont(STRETCH_TRANSFORM))
+                      .flatMap(StretchedFontTest::testFont)
+                      .filter(Objects::nonNull)
+                      .toList();
+
+        if (!errors.isEmpty()) {
+            errors.forEach(System.err::println);
+            throw new Error(errors.size() + " failure(s) found;"
+                            + " the first one: " + errors.get(0));
+        }
+    }
+
+    /**
+     * Tests the font with a set of text antialiasing hints.
+     *
+     * @param font the font to test
+     * @return a stream of test results
+     * @see #testFont(Font, Object)
+     */
+    private static Stream<String> testFont(final Font font) {
+        return Stream.of(VALUE_TEXT_ANTIALIAS_OFF,
+                         VALUE_TEXT_ANTIALIAS_ON,
+                         VALUE_TEXT_ANTIALIAS_LCD_HRGB)
+                     .map(hint -> testFont(font, hint));
+    }
+
+    /**
+     * Tests the font with the specified text antialiasing hint.
+     * In case of failure, it saves the rendered image to a file.
+     *
+     * @param font the font to test
+     * @param hint the text antialiasing hint to test
+     * @return {@code null} if the text rendered correctly; otherwise,
+     *         a {@code String} with the font family name and the value of
+     *         the rendering hint
+     */
+    private static String testFont(final Font font, final Object hint) {
+        final Dimension size = getTextSize(font);
+        final BufferedImage image =
+                new BufferedImage(size.width, size.height, TYPE_3BYTE_BGR);
```

Comment:
t might not be related but we can test the image with alpha as well, in that case I think the aa might be always disabled and/or some other code path will be executed.

The test for JDK-8204929 uses the "Locale.ENGLISH" and this one "Locale.US", does it matter which one is used to reproduce this bug?

Reply:
> It might not be related but we can test the image with alpha as well, in that case I think the aa might be always disabled and/or some other code path will be executed.

This test covers the disabled AA and greyscale AA.

Do you want to include both types of buffered images in the test? Do you want to switch to `TYPE_4BYTE_ABGR`? Do you want to use a translucent colour for rendering text?

> The test for JDK-8204929 uses the "Locale.ENGLISH" and this one "Locale.US", does it matter which one is used to reproduce this bug?

No, it doesn't matter here at all. Nothing would change if I didn't pass the locale, except that localised font names would be printed if a failure occurs.

I'll update it to `Locale.ENGLISH` for consistency with `GlyphRotationTest.java` from JDK-8204929.

Reply:
I think opaque+transparent images should be fine, do not think the actual format bgr/rgb/argb_pre/etc matters.